### ISSUE TITLE
jdkOnly = true turns off deepImmutablesDetection

### ISSUE THREAD

**Issue by narbsy** [immutables/immutables#642]
Hi,

I ran across a surprising bit of behavior today when trying to use both `jdkOnly` and `deepImmutablesDetection`.

Here's a boiled down example:

```
public interface DeepDetectionBroken {
    @Target({ElementType.PACKAGE, ElementType.TYPE})
    @Retention(RetentionPolicy.CLASS)
    @Value.Style(
            deepImmutablesDetection = true,
            jdkOnly = true)
    @interface DeepStyle {
    }

    @DeepStyle
    @Value.Immutable
    interface Deep {

    }

    @DeepStyle
    @Value.Immutable
    interface Container {
        Deep getDeep();
    }
}
```

With these classes, I would expect `ImmutableContainer` to have a field of `ImmutableDeep`, however what is generated instead is:

```
public final class ImmutableContainer implements DeepDetectionBroken.Container {
  private final DeepDetectionBroken.Deep deep;
  ...
}
```

If I remove `jdkOnly = true`, the expected is generated:

```
public final class ImmutableContainer implements DeepDetectionBroken.Container {
  private final ImmutableDeep deep;
  ...
}
```

Is this intentional behaviour?

Let me know if you need any additional details!

Thanks,
Chris

**Comment by elucash** [comment]
Thank you for reporting this. This is not intentional, most probably, or, if intentional, I cannot remember why it should. I'll try to find it out

**Comment by narbsy** [comment]
Thanks @elucash ! Let me know if you need anything else from me :)

**Comment by elucash** [comment]
Released in 2.5.5, thank you!
